site/content/contribute/more-info/mobile/e2e.md

```diff
@@ -0,0 +1,133 @@
+* `/detox/e2e/test` or {{< newtabref href="https://github.com/wix/Detox/blob/master/docs/APIRef.TestLifecycle.md" title="Test Files" >}}: To start writing tests, create a new file (e.g. `login.e2e.js`) in the `/detox/e2e/test` folder.
+    - The subfolder naming convention depends on the test grouping, which is usually based on the general functional area (e.g. `/detox/e2e/test/messaging/` for "Messaging").
+    - Test cases that require an Enterprise license should fall under `/detox/e2e/test/enterprise/`. This is to easily identify license requirements, both during local development and production testing for Enterprise features.
+* `/detox/.detoxrc.json`: for Detox {{< newtabref href="https://github.com/wix/Detox/blob/master/docs/APIRef.Configuration.md" title="configuration" >}}.
+* `/detox/package.json` : for all dependencies related to Detox end-to-end testing.
+
+### Writing an E2E Test
+This process has many similarities to [writing an E2E test for the mattermost-webapp project]({{<relref "contribute/more-info/webapp/e2e-testing.md">}}).
+Before writing a script, ensure that it has a corresponding test case in Zephyr. All test cases may be found in this {{< newtabref href="https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/design?projectId=10302" title="link" >}}. If test case is not available, feel free to prompt the QA team who will either search from an existing Zephyr entry or if it's a new one, it will be created for you.
+
+1. Create a test file based on the file structure aforementioned above.
+2. Include Zephyr identification (ID) and title in the test description, following the format of `it('[Zephyr_id] [title]')` or `it('[Zephyr_id]_[step] [title]')` if the test case has multiple steps. For test case "{{< newtabref href="https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T109" title="MM-T109 RN apps: User can't send the same message repeatedly" >}}", it should be:
+    ```javascript
+    describe('Messaging', () => {
+        it('MM-T109 User can\'t send the same message repeatedly', () => {
+            // Test steps and assertion here
+        }
+    }
+    ```
+4. Target an element using available {{< newtabref href="https://github.com/wix/Detox/blob/master/docs/APIRef.Matchers.md#matchers" title="matchers" >}}. For best results, it is recommended to match elements by unique identifiers using `testID`. The identifier should follow the following format to avoid duplication, `<location>.<modifier>.<element>.<identifier>`, where:
```

Comment:
```suggestion
4. Target an element using available matchers. For best results, it is recommended to match elements by unique identifiers using `testID`. The identifier should follow the following format to avoid duplication, `<location>.<modifier>.<element>.<identifier>`, where:
```